src/utils/ParseRouteUtils.js

```diff
@@ -512,14 +587,53 @@ function parseBusRoutes(busRoutes: Array<Object>, destinationName: string): Prom
         };
       }));
 
+      const startName = directions.length > 0 ? directions[0].name : null;
+      const endName = directions.length > 0 ? directions[directions.length - 1].name : null;
+
+      let previousStopName;
+      const routeSummary = directions.map((dir, index) => {
+        const { name, type, stayOnBusForTransfer } = dir;
+        // We want to ignore the intial walking direction
+        if (index === 0 && type === DIRECTION_TYPE.WALK) {
+          return null;
+        }
+        const direction = {
```

Comment:
define `direction` on one line

Reply:
this variable also seems to be named improperly, we're looping over the directions already, so isn't this not a `direction` object?

Reply:
it is still a direction object. this represents a `RouteSummary` object where `direction` is a simplified version of our current directions